tracer/src/Datadog.Trace.Tools.Runner/AnalyzeInstrumentationErrorsCommand.cs

```diff
@@ -0,0 +1,144 @@
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Tools.Runner.Checks;
+using Spectre.Console;
+using Spectre.Console.Cli;
+
+namespace Datadog.Trace.Tools.Runner;
+
+internal class AnalyzeInstrumentationErrorsCommand : AsyncCommand<AnalyzeInstrumentationErrorsSettings>
+{
+    public override async Task<int> ExecuteAsync(CommandContext context, AnalyzeInstrumentationErrorsSettings settings)
+    {
+        AnsiConsole.Record();
+        var process = $"'{settings.ProcessName ?? "na"}'";
+        if (settings.Pid != null)
+        {
+            process += ", ID: " + settings.Pid;
+        }
+
+        AnsiConsole.WriteLine("Running instrumentation error analysis on process " + process);
+        if (!string.IsNullOrEmpty(settings.Method))
+        {
+            AnsiConsole.WriteLine("Error was in method " + "." + settings.Method);
+        }
+
+        var logDirectory = GetLogDirectory(settings.LogDirectory, settings.Pid);
+        if (logDirectory == null)
+        {
+            Utils.WriteError("Log directory does not exist.");
+            return -1;
+        }
+
+        var processLogDir = GetProcessLogDirectory(logDirectory, settings);
+
+        if (!Directory.Exists(processLogDir))
+        {
+            Utils.WriteError("Log directory has not found.");
+            return -1;
+        }
+
+        var generatorArgs = new AssemblyGeneratorArgs(processLogDir, modulesToVerify: settings.Module == null ? null : new[] { settings.Module });
+
+        var exportedModulesPathsAndMethods = InstrumentedAssemblyGeneration.Generate(generatorArgs);
+
+        bool allVerificationsPassed = true;
+
+        foreach (var instrumentedAssembly in exportedModulesPathsAndMethods)
+        {
+            var result = new VerificationsRunner(instrumentedAssembly.InstrumentedAssemblyPath, instrumentedAssembly.OriginalAssemblyPath, instrumentedAssembly.InstrumentedMethods.Select(m => (m.MethodName, m.TypeName)).ToList()).Run();
+            AnsiConsole.WriteLine($"Verification for {instrumentedAssembly.InstrumentedAssemblyPath} {(result.IsValid ? "passed." : "failed: " + result.FailureReason)}");
+            allVerificationsPassed = allVerificationsPassed && result.IsValid;
+        }
+
+        try
+        {
+            foreach (var instrumented in exportedModulesPathsAndMethods)
+            {
+                foreach (var method in instrumented.InstrumentedMethods)
+                {
+                    AnsiConsole.WriteLine("--------------------------------------------------------");
+                    AnsiConsole.WriteLine($"Method {method.FullName} has been instrumented.");
+                    AnsiConsole.WriteLine($"Instrumented IL Code:{Environment.NewLine}{method.Instructions}");
+                    AnsiConsole.WriteLine($"Instrumented Decompiled Code:{Environment.NewLine}{method.DecompiledCode.Value}");
+                }
+            }
+
+            var checkProcessReport = AnsiConsole.ExportText();
```

Comment:
```suggestion
```
This was only needed for the instrumentation-telemetry piece, which will be added in a separate PR.

Reply:
Fixed in d6e34e581c70cc2271806c12db2c3d4f2caf78f1